sdk/mgmtcommon/Auth/Az.Auth/Az.Authentication/CertificateAuthenticationProvider.cs

```diff
@@ -65,7 +65,7 @@ public CertificateAuthenticationProvider(string certificateFilePath, bool IsCert
             _assertionProvider = (clientId) => Task.FromResult<ClientAssertionCertificate>(new ClientAssertionCertificate(clientId, certificateFilePath));
         }
 
-        public CertificateAuthenticationProvider(ClientAssertionCertificate certAssertion, bool IsCertRollOverEnabled) : this()
+        public CertificateAuthenticationProvider(IClientAssertionCertificate certAssertion, bool IsCertRollOverEnabled) : this()
```

Comment:
how the change fixing the problem? besides `ClientAssertionCertificate` any other impl of the interface `IClientAssertionCertificate`?

Reply:
For how, you can see the PR in `libraries-for-net`. We could use other implementation as parameter. The implementation is in the original repo in aad. I have commented above.